### PR TITLE
[9.x] Move function `withoutTrashed` into `DatabaseRule`

### PR DESCRIPTION
**Pull request by michaelnabil230**
ُThis PR move `withoutTrashed` into `DatabaseRule` makes more every sense

### CODE REVIEW

File: src/Illuminate/Validation/Rules/DatabaseRule.php [hunk 1]
```diff
@@ -169,6 +169,19 @@ public function whereNotIn($column, $values)
         });
     }
 
+    /**
+     * Ignore soft deleted models during the existence check.
```
Comment (amims71) [inline src/Illuminate/Validation/Rules/DatabaseRule.php:173]: I believe, it would work with unique check as well with the existence check.

**Comment by driesvints** [comment]
This is unfortunately a breaking change for third party packages that use these traits separately.

**Comment by michaelnabil230** [comment]
But there will be no problem with this change.

**Comment by michaelnabil230** [comment]
@driesvints 
But there will be no problem with this change.

**Comment by michaelnabil230** [comment]
@BrandonSurowiec This is what you said to me @driesvints, what do you think?

**Comment by driesvints** [comment]
People using the trait directly will see a method suddenly missing from the Exists rule. This is a breaking change.

**Comment by driesvints** [comment]
@michaelnabil230 feel free to re-attempt this

**Comment by michaelnabil230** [comment]
@driesvints It seems to me that you didn't see the all changes.
I merged `withoutTrashed' into '‏DatabaseRule'

**Comment by michaelnabil230** [comment]
@driesvints 
@BrandonSurowiec 

The new PR https://github.com/laravel/framework/pull/44938

**Pull request closed without merging**